lib/inferno/dsl/fhir_evaluator/evaluator.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'dotenv'
+require 'faraday'
+require 'fhir_models'
+require 'thor'
+require 'erb'
+require 'yaml'
+require 'pathname'
+
+require_relative 'config'
+require_relative 'rule'
+require_relative 'evaluation_context'
+require_relative 'evaluation_result'
+require_relative 'dataset_loader'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      class Evaluator
+        attr_accessor :data, :config
+
+        def initialize(data, config)
```

Comment:
I haven't kept up with all of @Jammjammjamm 's comments so I hope I'm not suggesting a reversion of one of them here, but the method signatures in this class are kind of reversed from how they originally were in the gitlab repo:

```ruby
    def initialize(the_ig)
      @ig = the_ig
    end

    def evaluate(data, summary = nil, config = Config.new)
```

At minimum the ig param will need to be added back in somewhere. 

But the way I originally envisioned this is that one could theoretically create one evaluator per IG, and then call evaluate on multiple sets of data. As it is now I guess I don't see the benefit of the constructor and the attr_accessors, everything is pretty contained in the evaluate method.

Reply:
I actually removed the IG part while refactoring `evaluator` as you took over the IG task and expected to handle whatever about IG. I will put IG attr_accessors back in constructor.

If the evaluator is designed to use for one IG with multiple data, Yes, data shouldn't be attr_accessors. About Config, for now it is fixed and universal for all data, but we only have minimum (almost empty) there. I will put it under `evaluate` as you suggested.